sklearn/metrics/_plot/precision_recall_curve.py

```diff
@@ -186,21 +226,75 @@ def plot(
         `drawstyle="default"`. However, the curve will not be strictly
         consistent with the reported average precision.
         """
-        self.ax_, self.figure_, name = self._validate_plot_params(ax=ax, name=name)
-
-        default_line_kwargs = {"drawstyle": "steps-post"}
-        if self.average_precision is not None and name is not None:
-            default_line_kwargs["label"] = (
-                f"{name} (AP = {self.average_precision:0.2f})"
+        name_ = name if name else self.curve_name
+        # If multi-curve, ensure all args are of the right length
+        req_multi = [
+            input for input in (self.precision, self.recall) if isinstance(input, list)
+        ]
+        if req_multi and ((len(req_multi) != 2) or len({len(arg) for arg in req_multi}) > 1):
+            raise ValueError(
+                "When plotting multiple precision-recall curves, `self.precision` "
+                "and `self.recall` should both be lists of the same length."
             )
-        elif self.average_precision is not None:
-            default_line_kwargs["label"] = f"AP = {self.average_precision:0.2f}"
-        elif name is not None:
-            default_line_kwargs["label"] = name
+        n_multi = len(self.precision) if req_multi else None
+        if req_multi:
+            for name, param in zip(
+                ["self.average_precision", "`name` or `self.curve_name`"],
+                (self.average_precision, name_)
+            ):
+                if not((isinstance(param, list) and len(param) != n_multi) or param is not None):
+                    raise ValueError(
+                        f"For multi precision-recall curves, {name} must either be "
+                        "a list of the same length as `self.precision` and "
+                        "`self.recall`, or None."
+                    )
```

Comment:
I struggled to come up with a nice way to do this. The checks we need are:

* `precision` and `recall`, both need to be lists of the same length or both need to be single ndarray
* for multi curve, `average_precision` and `name` can either be a list of the same length or None. 

This latter point is important, as previously I simply checked that all 4 parameters are of the same length if they were lists. I didn't check that 2 optional parameters needed to be `None` if they were not a list, for the multi-curve situation.

Suggestions welcome for making this nicer.

The good part though is that this is easily factorized out and can be generalised for all similar displays.